serializers/src/glTF/2.0/glTFExporter.ts

```diff
@@ -728,9 +729,28 @@ export class _Exporter {
                 break;
             }
             case VertexBuffer.ColorKind: {
+                const meshMaterial = (babylonTransformNode as Mesh).material;
+                const convertToLinear = meshMaterial ? (meshMaterial instanceof StandardMaterial) : true;
+                const tmpColor3 = TmpColors.Color3[0];
+                const tmpColor4 = TmpColors.Color4[0];
                 for (let k = 0, length = meshAttributeArray.length / stride; k < length; ++k) {
                     index = k * stride;
-                    const vertexData = stride === 3 ? Vector3.FromArray(meshAttributeArray, index) : Vector4.FromArray(meshAttributeArray, index);
+                    let vertexData;
+                    if (stride === 3) {
+                        vertexData = Vector3.FromArray(meshAttributeArray, index);
+                        if (convertToLinear) {
+                            Color3.FromArrayToRef(meshAttributeArray, index, tmpColor3);
+                            tmpColor3.toLinearSpace();
+                            vertexData.copyFromFloats(tmpColor3.r, tmpColor3.g, tmpColor3.b);

```

Comment:
Instead of the 3 lines above, I think you could simply do:
```typescript
vertexData.toLinearSpaceToRef(vertexData);
```
and get rid of the temporary colors.

Reply:
We don't always want to convert to linear space, if the input color is already expressed in linear, we'll convert again, and get the wrong color.

Reply:
Unfortunately we can't cast Vector3 to Color3...

Reply:
Unfortunately we can't cast Vector3 to Color3 with how we currently define the types.

Reply:
Can't you do:
```
vertexData = Color3.FromArray(meshAttributeArray, index);
```
instead of:
```
vertexData = Vector3.FromArray(meshAttributeArray, index);
```
to only work with colors?

Reply:
Thanks, I see what you mean now, I was able to avoid having to do an additional copy/conversion by following your suggestion.